graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,48 @@
+# Alternate Key Pattern
+
+Microsoft Graph API Design Pattern
+
+*The Alternate Key Pattern provides the ability to query for a single, specific entity identifiable through an alternative attribute (called key) that is not its unique identifier*
```

Comment:
Rather than "unique identifier", I think you should say "primary key".

Reply:
The reason I did not call it that way is that primary key conveys the idea that there is a database backing the system in question - which might not be the case: I might have an unique identifier for a 100% in memory system, or something else.

Am I making sense?

Reply:
Yeah - that makes sense.  The problem is that there is still a notion of a "primary" unique identifier and an "alternative" unique identifier.  The addressing patterns for the "primary" and the alternative are different too.

OK if you have an issue with this being too "database" backed (which I totally get), then maybe we need to change the filename and topic to be called "The alternate unique identifier pattern"?

Reply:
Thinking more about this, I believe you are right, and primary key is generic enough. For some reason my brain immediately linked in to a database situation, which is definitely not always the case. I have changed the wording in 1f3d11e